### ISSUE TITLE
REQUEST: Repository maintenance on opentelemetry-go* and opentelemetry-collector*

### ISSUE THREAD

**Issue by pellared** [open-telemetry/community#1926]
### Affected Repository

https://github.com/open-telemetry/opentelemetry-go
https://github.com/open-telemetry/opentelemetry-go-contrib
https://github.com/open-telemetry/opentelemetry-go-instrumentation
https://github.com/open-telemetry/opentelemetry-go-build-tools
https://github.com/open-telemetry/opentelemetry-proto-go
https://github.com/open-telemetry/opentelemetry-collector-contrib
https://github.com/open-telemetry/opentelemetry-collector

### Requested changes

> [!NOTE]  
> I think it may be more pragmatic to generate an organization-wide token in https://app.codecov.io/account/gh/open-telemetry/org-upload-token and add organization-wide GitHub secrets for GitHub Actions and Dependabot.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-go/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-go/settings/secrets/dependabot  with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-go/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-go-contrib/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-go-contrib/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-go-contrib/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-go-instrumentation/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-go-instrumentation/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-go-instrumentation/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-go-build-tools/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-go-build-tools/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-go-build-tools/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-proto-go/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-proto-go/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-proto-go/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-collector/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-collector/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-collector/settings.

Add `CODECOV_TOKEN` repository secret to https://github.com/open-telemetry/opentelemetry-collector-contrib/settings/secrets/actions and https://github.com/open-telemetry/opentelemetry-collector-contrib/settings/secrets/dependabot with value taken from https://app.codecov.io/gh/open-telemetry/opentelemetry-collector-contrib/settings.

### Purpose

There is a new release v4 of https://github.com/codecov/codecov-action
Take notice (from release notes):
> Tokenless uploading is unsupported. However, PRs made from forks to the upstream public repos will support tokenless (e.g. contributors to OS projects do not need the upstream repo's Codecov token). This [doc](https://docs.codecov.com/docs/adding-the-codecov-token#github-actions) shows instructions on how to add the Codecov token.
 
Slack thread: https://cloud-native.slack.com/archives/C01NJ7V1KRC/p1706772233505499

### Expected Duration

Permanently

### Repository Maintainers

@open-telemetry/go-maintainers 
@open-telemetry/collector-maintainers

**Comment by MrAlias** [comment]
Please include:

- https://github.com/open-telemetry/opentelemetry-go-contrib
- https://github.com/open-telemetry/opentelemetry-go-instrumentation
- https://github.com/open-telemetry/opentelemetry-go-build-tools
- https://github.com/open-telemetry/opentelemetry-proto-go

**Comment by pellared** [comment]
@MrAlias Done

**Comment by codeboten** [comment]
I can open a separate issue, but the same will be needed for the collector repositories 😬 

* https://github.com/open-telemetry/opentelemetry-collector-contrib
* https://github.com/open-telemetry/opentelemetry-collector

**Comment by pellared** [comment]
@codeboten Updated

**Comment by pellared** [comment]
I learned that for dependabot the secrets needs to be additionally added to "Dependabot secrets".

I think it may be more pragmatic to generate an organization-wide token in https://app.codecov.io/account/gh/open-telemetry/org-upload-token and add organization-wide GitHub secrets for GitHub Actions and Dependabot.

**Comment by MrAlias** [comment]
ping @open-telemetry/technical-committee

**Comment by jack-berg** [comment]
@open-telemetry/technical-committee - WDYT about this suggestion:

> I think it may be more pragmatic to generate an organization-wide token in https://app.codecov.io/account/gh/open-telemetry/org-upload-token and add organization-wide GitHub secrets for GitHub Actions and Dependabot.

I think its a good idea. Can go ahead an add the secret if folks agree.

**Comment by MrAlias** [comment]
Thank you @arminru. That looks to have unblocked us :tada:

**Comment by pellared** [comment]
@arminru `CODECOV_TOKEN` is missing as **Dependabot** organization-wide secret.

Because of this we cannot merge any of the following PRs created by Dependabot: https://github.com/open-telemetry/opentelemetry-go-build-tools/pulls/app%2Fdependabot

**Comment by arminru** [comment]
@pellared I added the same token also as a Dependabot org secret and triggered coverage builds for your open PRs again, now they're all passing.

**Comment by pellared** [comment]
@arminru Thanks. Closing.
